Sources/BarcodeHeroUI/Controllers/BHBaseCameraScanController.swift

```diff
@@ -0,0 +1,122 @@
+// Copyright © 2021 SpotHero, Inc. All rights reserved.
+
+#if !os(watchOS) && canImport(UIKit)
+    
+    import AVFoundation
+    import Foundation
+    import UIKit
+    
+    #warning("TODO: Make the controller work well in any orientation.")
+
+    // MARK: - Protocols
+
+    @available(iOS 9.0, *)
+    @available(tvOS, unavailable)
+    @available(watchOS, unavailable)
+    public protocol BHCameraScanControllerDelegate: AnyObject {
+        func didCapture(metadataObjects: [AVMetadataObject], from controller: BHBaseCameraScanController)
+    }
+
+    // MARK: - Classes
+
+    @available(iOS 9.0, *)
+    @available(tvOS, unavailable)
+    @available(watchOS, unavailable)
+    open class BHBaseCameraScanController: UIViewController {
+        fileprivate let session = AVCaptureSession()
+        private lazy var previewLayer = AVCaptureVideoPreviewLayer(session: self.session)
+        
+        fileprivate weak var delegate: BHCameraScanControllerDelegate?
+        
+        // MARK: Methods - Initializers
+        
+        public init(delegate: BHCameraScanControllerDelegate) {
+            super.init(nibName: nil, bundle: nil)
+            
+            self.delegate = delegate
+        }
+        
+        public required init?(coder aDecoder: NSCoder) {
+            super.init(coder: aDecoder)
+        }
+        
+        // MARK: Methods - Lifecycle
+        
+        override open func viewDidLoad() {
+            super.viewDidLoad()
+            
+            if let device = AVCaptureDevice.default(for: .video),
```

Comment:
all the `AVFoundation` code in this file was a copy/paste from the previous implementation.